### PR TITLE
chore: make module loaders DML aware and auto generate joiner config

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
When a module is automatically loaded from its location, if the module service does not provides us with a joiner config then we will auto generate the joiner config based on the models that are present in the module no matter if they are mikro orm entities or dml entities.

This pr also allow a module to load with dml entities as models.

This pr also include refactoring for the `Modules` usage to cleanup inconsistencies and always import it from the `utils` package.

Also, now the DML export `model` and `toMikroOrmEntity` utilities

**Note**

This auto loading mechanism only works if we load the resources ourselves from a path, which will likely be all cases, if someone provides the module exports directly then they ll have to handle resources manually but i think it is fine

FIXES CORE-2353
FIXES CORE-2339

### CODE REVIEW

File: packages/core/utils/src/dml/entity.ts [hunk 1]
```diff
@@ -13,9 +13,25 @@ import {
...
+   * A static method to check if an entity is an instance of DmlEntity.
+   * It allows us to identify a specific object as being an instance of
+   * DmlEntity.
+   *
+   * @param entity
+   */
+  static isDmlEntity(entity: any): entity is DmlEntity<any> {
+    return (
+      entity instanceof DmlEntity ||
+      entity.__dmlEntitySymbol__ === this.__dmlEntitySymbol__
```
Comment (thetutlage) [inline packages/core/utils/src/dml/entity.ts]: Will `__dmlEntitySymbol__` exist on the instance?

Comment (adrien2p) [reply packages/core/utils/src/dml/entity.ts]: yes I fixed it in my commit ahah

**Comment by adrien2p** [comment]
The review focus here can be on `modules-sdk` and `utils` package, I havent migrated the modules to rely on the joiner config autogeneration and they are not yet using the DML

**Review by riqwan — APPROVED** [review]
nice!

**Review by olivermrbl — APPROVED** [review]
Nice work and cleanup!

**Review by sradevski — APPROVED** [review]
Nice 👍

**Comment by olivermrbl** [comment]
@adrien2p the starter will need to be updated to import `Modules` from utils as well. This is what is causing the CI to fail

**Comment by adrien2p** [comment]
@olivermrbl the v2 branch of the starter already import from the utils

**Comment by olivermrbl** [comment]
Ah sorry, myb ad!

**Comment by olivermrbl** [comment]
@adrien2p, I believe it hasn't been updated actually: https://github.com/medusajs/medusa-starter-default/tree/feat/v2-ci

**Comment by adrien2p** [comment]
> @adrien2p, I believe it hasn't been updated actually: https://github.com/medusajs/medusa-starter-default/tree/feat/v2-ci

@olivermrbl from what I ve seen it is importing from utils, it is not new that the modules enum is exported from utils :), but our packages were inconsistent

**Comment by adrien2p** [comment]
@olivermrbl shall we? 😊

**Comment by olivermrbl** [comment]
Let's do it!

**Pull request merged**